okapi-core/src/main/java/org/folio/okapi/util/ProxyContext.java

```diff
@@ -237,7 +237,10 @@ public void responseError(ErrorType t, Throwable cause) {
   }
 
   private void responseError(int code, Throwable cause) {
-    if (cause != null && cause.getMessage() != null) {
+    if (code == 500) {
+      logger.warn(cause.getMessage(), cause);
+    }
+    if (cause.getMessage() != null) {
```

Comment:
cause can be null
`public void responseError(ErrorType t, Throwable cause)` might be called with a null cause and calls `private void responseError(int code, Throwable cause) `.

    String msg = (cause != null && cause.getMessage() != null) ? cause.getMessage() : messages.getMessage("10300");
    if (code == 500) {
      logger.warn(msg, cause);
    }
    responseError(code, msg);

Reply:
I don't see how cause can be null, so can not make coverage of that, unless I fake it with a ProxyContext unit test.

Reply:
Regret that I didn't make this change in a separate PR. It an important backtrace to get..